### ISSUE TITLE
[v4] Blocks field: max option ignored when pasting blocks

### ISSUE THREAD

**Issue by SebastianEberlein-JUNO** [k-next/kirby#5670]
## Description
I set the `max` option in a blocks field:

```yml
singleBlock:
  label: Just 1 block
  type: blocks
  max: 1
```

When I open the block options and click on "Paste after", I can add more than 1 block.

https://github.com/getkirby/kirby/assets/77532479/519e3780-3f0d-449d-9c8c-23397a7aa5d1

**Expected behavior**  
In Kirby 3, pasting blocks was not possible if the `max` number of blocks was reached.

## Your setup
v4-beta.1

**Your system (please complete the following information)**
 - OS: macOS
 - Browser: Chrome

**Comment by afbora** [comment]
### Summary

The issue is that the `max` check of the pasted blocks is done [in the backend.](https://github.com/getkirby/kirby/blob/4.0.0-beta.1/src/Form/Field/BlocksField.php#L158-L161) If `max: 1`, if one block is already added, the backend does not know how many blocks are in the frontend, so it takes only one from the copied block list and sends it to the frontend. There was already one block in the panel and one more is added.

In short, the `max` control is applied for the pasted blocks, not for current blocks/value in the block.
